sdks/java/core/src/main/java/org/apache/beam/sdk/io/WriteFiles.java

```diff
@@ -996,10 +1001,18 @@ public void finishBundle(FinishBundleContext c) throws Exception {
           c.output(result.getValue(), result.getKey(), result.getValue().getWindow());
         }
       } finally {
-        deferredOutput = null;
-        closeFutures = null;
+        deferredOutput.clear();
+        closeFutures.clear();
       }
     }
+
+    // Ensure that transient fields are initialized.
```

Comment:
I haven't seen this before, is this preferrable to just initializing these in class constructor?

Reply:
Class constructor isn't involved during deserialization of the object and transient fields are always initialized to their defaults (e.g. int = 0, boolean = false, object = null, ...). This allows us to initialize the fields during deserialization to be a non null object.

See https://docs.oracle.com/javase/7/docs/api/java/io/Serializable.html for more details.

Reply:
Any reason to prefer this to the @Setup method?
If not, I'd say that is preferrable since it is a standard Beam DoFn method and the name is more self-explanatory.

Reply:
I forgot about `@Setup`. Swapped to using it.

Reply:
And reverted back because spot bugs prefers it during deserialization so that we can ensure that the fields are set regardless since spotbugs doesn't know about the DoFn lifecycle.